File: java/01-beginner/01-basics/HelloWorld.java
/**
 * LESSON: Your First Java Program
 * Every Java program needs a class and a main method.
 */
public class HelloWorld {
    public static void main(String[] args) {
        // println prints text and moves to the next line
        System.out.println("Hello, World!");
        System.out.println("Welcome to Java Programming!");

        // print prints text WITHOUT moving to the next line
        System.out.print("Hello ");
        System.out.print("World ");
        System.out.println(); // just prints a new line

        // printf allows formatted output (like C)
        String name = "Emmanuel";
        int age = 20;
        System.out.printf("My name is %s and I am %d years old%n", name, age);
    }
}

File: cpp-practice/01-beginner/01-basics/hello_world.cpp
/**
 * LESSON: Your First C++ Program
 * Every C++ program needs a main() function as the entry point.
 * #include brings in library headers.
 *
 * Compile: g++ -o hello hello_world.cpp
 * Run:     ./hello
 */
#include <iostream>  // for cout, cin, endl

int main() {
    // cout (character output) prints to the console
    // << is the insertion operator
    std::cout << "Hello, World!" << std::endl;

    // endl flushes the buffer and adds newline
    // '\n' is faster (no flush) and usually preferred
    std::cout << "Using endl" << std::endl;
    std::cout << "Using \\n\n";

    // Printing multiple values
    std::string name = "Emmanuel";
    int age = 20;
    std::cout << "Name: " << name << ", Age: " << age << "\n";

    // printf-style formatting (from C)
    printf("Formatted: %s is %d years old\n", name.c_str(), age);

    // return 0 means the program ran successfully
    return 0;
}
